### PR TITLE
dnn/openvino: support run inference via GPU

### PR DESCRIPTION
**Pull request by Semmer2**

### CODE REVIEW

File: libavfilter/dnn/dnn_backend_openvino.c [hunk 1]
```diff
@@ -41,14 +49,15 @@ typedef struct OVModel{
...
 } OVModel;
 
-static const AVClass dnn_openvino_class = {
-    .class_name = "dnn_openvino",
-    .item_name  = av_default_item_name,
-    .option     = NULL,
-    .version    = LIBAVUTIL_VERSION_INT,
-    .category   = AV_CLASS_CATEGORY_FILTER,
+#define OFFSET(x) offsetof(OVContext, x)
+#define FLAGS AV_OPT_FLAG_FILTERING_PARAM
```
Comment (Semmer2) [inline libavfilter/dnn/dnn_backend_openvino.c:57]: As I noticed, the AV_OPT_FLAG_VIDEO_PARAM is only used in show help of "./ffmpeg -h full" (fftools/cmdtuils.c: show_hep_children), to show all filters details. While the cmd will not show the options of OpenVINO options, so should I omitted it as above or try to code to show the openvino options in ffmpeg help?

**Comment by Semmer2** [comment]
all issues mentioned in last version patch have been modified.

File: libavfilter/dnn/dnn_backend_openvino.c (hunk 1 above)
Comment (guoyejun) [reply libavfilter/dnn/dnn_backend_openvino.c:57]: just ignore now. It is not designed for our case.

File: libavfilter/dnn/dnn_backend_openvino.c [hunk 2]
```diff
@@ -159,10 +168,18 @@ static DNNReturnType set_input_ov(void *model, DNNData *input, const char *input
...
     OVModel *ov_model = NULL;
+    OVContext *ctx = NULL;
     IEStatusCode status;
     ie_config_t config = {NULL, NULL, NULL};
+    ie_available_devices_t *a_dev;
+    a_dev = av_malloc(sizeof(ie_available_devices_t));
+    if (!a_dev){
+        av_log(NULL, AV_LOG_ERROR, "Failed to allocate memory for device info\n");
+        return NULL;
+    }
```
Comment (guoyejun) [inline libavfilter/dnn/dnn_backend_openvino.c]: why need check here

File: libavfilter/dnn/dnn_backend_openvino.c [hunk 3]
```diff
@@ -173,19 +190,46 @@ DNNModel *ff_dnn_load_model_ov(const char *model_filename, const char *options)
...
     if (status != OK)
         goto err;
 
-    status = ie_core_load_network(ov_model->core, ov_model->network, "CPU", &config, &ov_model->exe_network);
+    //parse options
+    if (av_opt_set_from_string(ctx, options, NULL, "=", "&") < 0) {
+        av_log(ctx, AV_LOG_ERROR, "Failed to parse options \"%s\"\n", options);
+        goto err;
+    }
+    if (!ctx->options.device_type) {
```
Comment (guoyejun) [inline libavfilter/dnn/dnn_backend_openvino.c]: don't check device_type here, we just pass it to ie_core_load_network. If it returns error, then, we list all the supported devices.   

Let's talk F2F if necessary.

**Pull request closed without merging**